devito/symbolics/manipulation.py

```diff
@@ -319,12 +319,88 @@ def reuse_if_untouched(expr, args, evaluate=False):
         return expr.func(*args, evaluate=evaluate)
 
 
-def evalmin(a, b):
+def evalrel(func=min, input=None, assumptions=None):
     """
-    Simplify min(a, b) if possible.
+    The purpose of this function is two-fold: (i) to reduce the `input` candidates of a
+    for a MIN/MAX expression based on the given `assumptions` and (ii) return the nested
+    MIN/MAX expression of the reduced-size input. For this purpose this function takes
+    advantage of Python's built-in `min`, SymPy's `Min` and our in-house `MIN`.
+
+    Parameters
+    ----------
+    func : min or max, the operation to follow
+        Defaults to min
+    input : a list of the candidate symbols to be simplified
+        The candidates to be evaluated. Defaults to None.
+    assumptions : A list of assumptions formed as relationals between candidates,
+        assumed to be True. Defaults to None.
+
+    Examples
+    --------
+    Assuming no values are known for `a`, `b`, `c`, `d` but we know that `d<=a` and
+    `c>=b` we can safely drop `a` and `c` from the candidate list.
+
+    >>> from devito import Symbol
+    >>> a = Symbol('a')
+    >>> b = Symbol('b')
+    >>> c = Symbol('c')
+    >>> d = Symbol('d')
+    >>> evalrel(max, [a, b, c, d], [[Le(d, a), Ge(c, b)]])
+    MAX(a, c)
     """
+    sfunc = (Min if func is min else Max)  # Choose SymPy's Min/Max
+
+    # Form relationals so that RHS has more arguments than LHS:
+    # i.e. (a + d >= b) ---> (b <= a + d)
+    assumptions = [a if len(a.rhs.args) > len(a.lhs.args) else
+                   a.reversed for a in as_list(assumptions)]
+
+    # Expand relations if possible
+    processed = []
+    for a in as_list(assumptions):
+        if isinstance(a, (Ge, Gt)) and a.rhs.is_Add and a.lhs.is_positive:
+            # If `c >= a + b, {a, b, c} >= 0` then add 'c>=a, c>=b'
+            if all(i.is_positive for i in a.rhs.args):
+                processed.extend(Ge(a.lhs, i) for i in a.rhs.args)
+            # If `c >= a + b, a>= 0, b<=0` then add 'c>=b, c<=a'
+            elif len(a.rhs.args) == 2:
+                processed.extend(Lt(a.lhs, i) if i.is_positive else
+                                 Gt(a.lhs, i) for i in a.rhs.args)
+            else:
+                processed.append(a)
+        else:
+            processed.append(a)
+
+    # Symmetry/ Antisymmetry
```

Comment:
can you remind us what you're trying to accomplish here exactly? (just reply here with an example)

Reply:
Maybe docs are confusing here. Symmetry is just taken care at the elif. Nothing really special happening.